src/test/tv/codely/scala_intro_examples/lesson_09_oop/exercise_solutions/joan_miralles_solution/EmailTest.scala

```diff
@@ -0,0 +1,46 @@
+package tv.codely.scala_intro_examples.lesson_09_oop.exercise_solutions.joan_miralles_solution
+
+import java.security.InvalidParameterException
```

Comment:
Why [`java.security.InvalidParameterException`](https://docs.oracle.com/javase/7/docs/api/java/security/InvalidParameterException.html)?
>This exception, *designed for use by the JCA/JCE engine classes*, is thrown when an invalid parameter is passed to a method

Maybe [`java.lang.IllegalArgumentException`](https://docs.oracle.com/javase/8/docs/api/java/lang/IllegalArgumentException.html) would be more suitable for this case 🙂
>Thrown to indicate that a method has been passed an illegal or inappropriate argument.